On Ubuntu 12.04: _graphviz.so: undefined symbol: Agundirected

I've done the following on my dev hosts for a while now. I needed to bring up pygraphviz on a somewhat-freshly imaged host, and got a new error that I've not seen before when I tried to import (see last output; install seems to be smooth):

$ dpkg -s graphviz-dev
Package: graphviz-dev
Status: install ok installed
Priority: optional
Section: devel
Installed-Size: 90
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Architecture: all
Source: graphviz
Version: 2.26.3-10ubuntu1.2
Depends: libgraphviz-dev

v$ pip install pygraphviz
Downloading/unpacking pygraphviz
  Downloading pygraphviz-1.3.1.tar.gz (103kB): 103kB downloaded
  Running setup.py (path:/home/elis/venv/mobius3d_dev/build/pygraphviz/setup.py) egg_info for package pygraphviz
    
    warning: no previously-included files matching '*~' found anywhere in distribution
    warning: no previously-included files matching '*.pyc' found anywhere in distribution
    warning: no previously-included files matching '.svn' found anywhere in distribution
    no previously-included directories found matching 'doc/build'
Installing collected packages: pygraphviz
  Running setup.py install for pygraphviz
    Trying pkg-config
    include_dirs=/usr/include/graphviz
    library_dirs=
    
    warning: no previously-included files matching '*~' found anywhere in distribution
    warning: no previously-included files matching '*.pyc' found anywhere in distribution
    warning: no previously-included files matching '.svn' found anywhere in distribution
    no previously-included directories found matching 'doc/build'
    building 'pygraphviz._graphviz' extension
    gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I/usr/include/graphviz -I/usr/include/python2.7 -c pygraphviz/graphviz_wrap.c -o build/temp.linux-x86_64-2.7/pygraphviz/graphviz_wrap.o
    pygraphviz/graphviz_wrap.c: In function ‘agattr_label’:
    pygraphviz/graphviz_wrap.c:3321:5: warning: return makes integer from pointer without a cast [enabled by default]
    pygraphviz/graphviz_wrap.c: In function ‘_wrap_agread’:
    pygraphviz/graphviz_wrap.c:3408:9: warning: unused variable ‘mode1’ [-Wunused-variable]
    pygraphviz/graphviz_wrap.c:3407:13: warning: unused variable ‘mode_byte_obj1’ [-Wunused-variable]
    pygraphviz/graphviz_wrap.c:3406:13: warning: unused variable ‘mode_obj1’ [-Wunused-variable]
    pygraphviz/graphviz_wrap.c:3405:7: warning: unused variable ‘fd1’ [-Wunused-variable]
    pygraphviz/graphviz_wrap.c: In function ‘_wrap_agwrite’:
    pygraphviz/graphviz_wrap.c:3466:9: warning: unused variable ‘mode2’ [-Wunused-variable]
    pygraphviz/graphviz_wrap.c:3465:13: warning: unused variable ‘mode_byte_obj2’ [-Wunused-variable]
    pygraphviz/graphviz_wrap.c:3464:13: warning: unused variable ‘mode_obj2’ [-Wunused-variable]
    pygraphviz/graphviz_wrap.c:3463:7: warning: unused variable ‘fd2’ [-Wunused-variable]
    gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro build/temp.linux-x86_64-2.7/pygraphviz/graphviz_wrap.o -L -lcgraph -lcdt -o build/lib.linux-x86_64-2.7/pygraphviz/_graphviz.so
Successfully installed pygraphviz
Cleaning up...

$ python -c 'import pygraphviz'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/elis/venv/mobius3d_dev/local/lib/python2.7/site-packages/pygraphviz/__init__.py", line 58, in <module>
    from .agraph import AGraph, Node, Edge, Attribute, ItemAttribute, DotError
  File "/home/elis/venv/mobius3d_dev/local/lib/python2.7/site-packages/pygraphviz/agraph.py", line 26, in <module>
    from . import graphviz as gv
  File "/home/elis/venv/mobius3d_dev/local/lib/python2.7/site-packages/pygraphviz/graphviz.py", line 28, in <module>
    _graphviz = swig_import_helper()
  File "/home/elis/venv/mobius3d_dev/local/lib/python2.7/site-packages/pygraphviz/graphviz.py", line 24, in swig_import_helper
    _mod = imp.load_module('_graphviz', fp, pathname, description)
ImportError: /home/elis/venv/mobius3d_dev/local/lib/python2.7/site-packages/pygraphviz/_graphviz.so: undefined symbol: Agundirected

--- a/setup_commands.py
+++ b/setup_commands.py
@@ -45,8 +45,10 @@ def add_extensions(command_subclass):
         if self.distribution and self.distribution.ext_modules:
             for m in self.distribution.ext_modules:
                 if m.name == 'pygraphviz._graphviz':
-                    m.include_dirs.append(self.include_path)
-                    m.library_dirs.append(self.library_path)
+                    if self.include_path:
+                        m.include_dirs.append(self.include_path)
+                    if self.library_path:
+                        m.library_dirs.append(self.library_path)
         orig_run(self)
 
     command_subclass.__init__ = __init__

